src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java

```diff
@@ -0,0 +1,166 @@
+                + missingArguments.toString(), 0, missingArguments.size());
+    }
+
+    @When("^In MongoDb RepeatFlagHandler collection the subscription has matched the AggrObjectId$")
+    public void in_MongoDb_RepeatFlagHandler_and_subscription_collections_the_subscription_has_matched_the_AggrObjectId() throws IOException {
+        processSubscription(subscriptionStrWithOneMatch, subscriptionWithOneMatch);
+        List<String> resultRepeatFlagHandler = mongoDBHandler.find(dataBaseName, repeatFlagHandlerCollection, subscriptionIdMatchedAggrIdObjQuery);
+        assertEquals(1, resultRepeatFlagHandler.size());
+        assertEquals("\"" + AGGREGATED_OBJECT_ID + "\"", getAggregatedObjectId(resultRepeatFlagHandler, 0));
+    }
+
+    @Then("^I make a DELETE request with subscription name \"([^\"]*)\" to the subscription REST API \"([^\"]*)\"$")
+    public void i_make_a_DELETE_request_with_subscription_name_to_the_subscription_REST_API(String name, String subscriptionEndPoint) throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.delete(subscriptionEndPoint + name).accept(MediaType.APPLICATION_JSON)).andReturn();
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+    }
+
+    @Then("^Check in MongoDB RepeatFlagHandler collection that the subscription has been removed$")
+    public void check_in_MongoDB_RepeatFlagHandler_collections_that_the_subscription_has_been_removed() throws IOException {
+        List<String> resultRepeatFlagHandler = mongoDBHandler.find(dataBaseName, repeatFlagHandlerCollection, subscriptionIdMatchedAggrIdObjQuery);
+        assertEquals("[]", resultRepeatFlagHandler.toString());
+        assertTrue(mongoDBHandler.dropDocument(dataBaseName, collectionName, getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH).toString()));
+    }
+
+    @When("^In MongoDb RepeatFlagHandler collection the subscription has matched the AggrObjectId at least two times$")
+    public void in_MongoDb_RepeatFlagHandler_collection_the_subscription_has_matched_the_AggrObjectId_at_least_two_times() throws IOException {
+        processSubscription(subscriptionStrWithTwoMatch, subscriptionWithTwoMatch);
+        List<String> resultRepeatFlagHandler = mongoDBHandler.find(dataBaseName, repeatFlagHandlerCollection, subscriptionIdMatchedAggrIdObjQuery);
+        assertEquals(2, resultRepeatFlagHandler.size());
+        assertEquals("\"" + AGGREGATED_OBJECT_ID + "\"", getAggregatedObjectId(resultRepeatFlagHandler, 0));
+        assertEquals("\"" + AGGREGATED_OBJECT_ID + "\"", getAggregatedObjectId(resultRepeatFlagHandler, 1));
+    }
+
+    @Override
+    public List<String> getEventNamesToSend() {
+        List<String> eventNames = new ArrayList<>();
+        eventNames.add("event_EiffelTestCaseFinishedEvent_2");
+        eventNames.add("event_EiffelActivityFinishedEvent");
+        return eventNames;
+    }
+
```

Comment:
You can add javadoc to the two functions below.

Reply:
added